OracleManagementAgent/README.md

```diff
@@ -0,0 +1,118 @@
+<!--
+    Copyright (c) 2022, Oracle and/or its affiliates.
+    Licensed under the Universal Permissive License v 1.0 as shown at
+    https://oss.oracle.com/licenses/upl.
+-->
+Oracle Management Agent Container Image
+=====
+This repository contains sample container configurations to facilitate installation and environment setup for DevOps users. This project includes Dockerfiles based on Oracle Linux and Oracle OpenJDK 8.
+
+The certification of Oracle Management Agent in a container does not require the use of any file presented in this
+repository. Customers and users are welcome to use them as starters, and customize/tweak, or create
+from scratch new scripts and Dockerfiles.
+
+
+## How to build and run
+
+Oracle Management Agent images use the Oracle Linux 7 Slim container image as the base image.
+
+#### Build steps
+
+1. [Download the Management Agent software](https://cloud.oracle.com/macs)
+
+    **Note: Select 'Downloads and Keys' then download 'Agent for LINUX (X86_64)' of the package type ZIP.**
+
+1. Copy the downloaded bundle to the same directory as the Dockerfile
+
+    ```shell
+    $ cp oracle.mgmt_agent.zip OracleManagementAgent/dockerfiles/latest/
+    ```
+
+1. Change directory to build the dockerfile
+
+    ```shell
+    $ cd OracleManagementAgent/dockerfiles/latest/
+    ```
+
+1. Use Docker to build a container image
+
+    ```shell
+    $ docker build -t oracle/mgmtagent-container .
+    ```
+
+1. [Create and Download the Install Key](https://docs.oracle.com/en-us/iaas/management-agents/doc/install-management-agent-chapter.html)
+
+    **Note: Follow the steps in the 'Create Install Key' and 'Configure a Response File' sections to create the install key and save it as input.rsp.**
+
+1. Create a Docker volume to share configs with the container
+
+    ```shell
+    $ docker volume create mgmtagent-volume
+
+    # identify the mount point location to use in next steps
+    $ docker volume inspect mgmtagent-volume|grep Mountpoint
+        "Mountpoint": "/var/lib/docker/volumes/mgmtagent-volume/_data",
+    ```
+
+1. Copy the Install Key (input.rsp) into the shared Docker volume Mountpoint
+
+    ```shell
+    # create any necessary dirs
+    $ mkdir -p /var/lib/docker/volumes/mgmtagent-volume/_data/mgmtagent_secret
+    $ cp input.rsp /var/lib/docker/volumes/mgmtagent-volume/_data/mgmtagent_secret/
+    ```
```

Comment:
Is the `input.rsp` required for first run or every run? This is not best practice when it comes to sharing content into a container. You should either use secrets (if the `input.rsp` contains credentials) or use a first-run script that obtains those secrets and stores them in the container.

Reply:
`input.rsp` is required up until the agent is registered. This is a runtime behavior dependent on what causes a registration failure, it can range from tenancy not configured correctly to any input misconfigured in the `input.rsp`. I take your point and have added docker-compose based container build and run execution. Docker-compose creates a secret and makes `input.rsp` available to the container without requiring additional user actions and hides the entire workflow.